chroma_core/models/host.py

```diff
@@ -1021,9 +1021,10 @@ def run(self, kwargs):
         old_session_id = AgentRpc.get_session_id(host.fqdn)
         self.invoke_agent(host, "restart_agent")
         AgentRpc.await_restart(host.fqdn, timeout=settings.AGENT_RESTART_TIMEOUT, old_session_id=old_session_id)
-        self.invoke_rust_local_action_expect_result(
-            "await_next_session", (host.fqdn, old_rust_session_id, settings.AGENT_RESTART_TIMEOUT)
-        )
+        if old_rust_session_id:
```

Comment:
I'm guessing you hit a case where this was `None`?

Reply:
I'm not sure if this is related but i've noticed if I remove a server and then add it back it doesn't have the same session and will not allow me to add the server. Is this related to that?